lib/cocoapods-core/dependency.rb

```diff
@@ -196,9 +201,9 @@ def compatible?(other)
     def ==(other)
       self.class == other.class &&
         name == other.name &&
-        requirement == other.requirement &&
         external_source == other.external_source &&
-        podspec_repo == other.podspec_repo
+        podspec_repo == other.podspec_repo &&
+        requirement == other.requirement
```

Comment:
`requirement` is a (relatively) heavier accessor than the others - short circuiting here improves runtime